docs/reference/docs/reindex.asciidoc

```diff
@@ -489,6 +489,10 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=wait_for_active_shards
 (Optional, Boolean) If `true`, the request blocks until the operation is complete.
 Defaults to `true`.
 
+`ignore_unavailable`::
+(Optional, Boolean) If false, the request returns an error if it targets a missing or closed index. Defaults to false.
+NOTE:: Should be set to `true` to perform reindexing from remote when index name is prefixed with remote cluster alias e.g. `cluster_name:index_name`
+
```

Comment:
I don't think this is a reasonable interface for the reindex API.

As it's possible to have a multiple reindex sources in the same reindex request, it could be that local and remote sources are mixed in the same request. In this case it's reasonable to support `ignore_unavailable=false` (which is the default), at least for the local indices (though I would prefer we also support it for remote indices too, which is how the search endpoint works, I think).

I noticed that the search request honours either `ignore_unvailable` options when using remote clusters (CCS).
Reindex uses search under the hood, so it should work, in principle... But, from what I can tell, it's the `ReindexValidator#validateAgainstAliases` check that trips it up. This check ought to prevent reindexing back onto the same index via an alias. I think it's this check that we need to adapt.

Is a remote index allowed as the **destination**? If not, and remote indices are only allowed as sources, then I think the easy fix is to exclude remote names from the list that the `validateAgainstAliases` check runs over, because it's not possible that a remote index name refer to a local index name.

Reply:
No, seems remote index is **NOT** allowed as destination